src/lib.rs

```diff
@@ -497,9 +490,11 @@ fn compress_parents_parallel(
     let mut parents_exact = child_chaining_values.chunks_exact(BLOCK_LEN);
     // Use MAX_SIMD_DEGREE_OR_2 rather than MAX_SIMD_DEGREE here, because of
     // the requirements of compress_subtree_wide().
-    let mut parents_array = ArrayVec::<[&[u8; BLOCK_LEN]; MAX_SIMD_DEGREE_OR_2]>::new();
-    for parent in &mut parents_exact {
-        parents_array.push(array_ref!(parent, 0, BLOCK_LEN));
+    let mut parents_array = [&[0u8; BLOCK_LEN]; MAX_SIMD_DEGREE_OR_2];
+    let mut parents_so_far = 0;
+    for (a, b) in parents_array.iter_mut().zip(&mut parents_exact) {
+        *a = array_ref!(b, 0, BLOCK_LEN);
+        parents_so_far += 1;
     }
     platform.hash_many(
         &parents_array,
```

Comment:
`parents_array` is not guaranteed to be `MAX_SIMD_DEGREE_OR_2` here; it can range anywhere from 2 to `MAX_SIMD_DEGREE`, so you need to pass the actual size somehow.

Reply:
This is definitely the cause of at least some of the test failures.

Reply:
@sneves Okay.. That's odd. I did try adding `[..parents_so_far]` after `&parents_array` and that didn't work... Now it does. Maybe a different write was skipped by accident somewhere else. *shrugs*. Thanks.